Create a means for caching task registration and request building. Ad…

inf3rnus opened the pull request:
Hey all, here's the caching proposal for task registration and request building.

Caching is optional, and has rewrite capabilities for task building and for task registration.

It works by using pickles, and by default all pickles are stored in lm_harness/caching/.cache. You can specify an alternate path via the env var `LM_HARNESSS_CACHE_PATH`. Which is really useful for docker containers.

I also added some tqdm's to track the progress of these steps.

In addition to this, I extended simple_evaluate to allow for you to take in a model_arg_dict, and added a class method to the Model class to allow for this. Passing a bunch of strings was a frustrating DX.

I also added some type annotations, and fixed a bug with the limit, where it doesn't return results if you set your float really low. At least one item in a dataset is run now if you set limit super low.

This is a draft, so if something's off please LMK, but would love to merge this puppy! (Hence the test script, I'll clean up the commit.)

Best,
Aaron

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=1372) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=1372) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=1372) it.</sub>

haileyschoelkopf commented on lm_eval/api/model.py:
```diff
@@ -133,6 +133,28 @@ def create_from_arg_string(
         return cls(**args, **args2)
 
+    @classmethod
+    def create_from_arg_obj(
```
This is a really nice addition, thanks!

inf3rnus replied:
Thank you!

haileyschoelkopf commented on lm_eval/caching/cache.py:
```diff
@@ -0,0 +1,34 @@
+
+MODULE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+OVERRIDE_PATH = os.getenv("LM_HARNESSS_CACHE_PATH")
```
```suggestion
OVERRIDE_PATH = os.getenv("LM_HARNESS_CACHE_PATH")
```

haileyschoelkopf commented on lm_eval/evaluator.py:
```diff
@@ -20,17 +22,24 @@
     model,
     model_args=None,
-    tasks=None,
+    model_arg_dict=None,
```
could we have `model_args` simply be of either type `str` or `dict` and change behavior based on that?

inf3rnus replied:
Absolutely, could benefit from some polymorphism, at a second glance, that makes a lot of sense, I'll make the change

haileyschoelkopf commented on test_script.py:
```diff
@@ -0,0 +1,94 @@
+
+# Used to specify alternate cache path, useful if run in a docker container
+# NOTE raw datasets will break if you try to transfer the cache from your host to an image
+LM_HARNESSS_CACHE_PATH = os.getenv("LM_HARNESSS_CACHE_PATH")
```
3 S's here instead of 2

haileyschoelkopf commented on lm_eval/tasks/__init__.py:
```diff
@@ -26,6 +26,9 @@
     QMSum,
 )
 
+from lm_eval.caching.cache import save_to_cache, load_from_cache
```
This file is getting a pretty heavy rewrite in #1321 -- would you be willing to check out shifting these changes to that once that settles?

#1321 should speed up task loading / registration but caching them would still be great.

inf3rnus replied:
Yeah for sure! np np. 

Only pushback would be that users who could benefit from the caching would be missing out on it in the interim, so if it's going to be a while, I'd lean towards us integrating the current changes, but if not, then strong agree 👍👍, def seems like it'll be a merge pain

haileyschoelkopf replied:
Let me get back to you on the timeline for that PR—think we were originally hoping to merge it this week but will check on that.

haileyschoelkopf replied:
We've merged that PR, as it should speed up task registry at initialization quite a bit in the meantime!

inf3rnus replied:
Sick! Yeah, I'll slice off a chunk of time this weekend and sharpen this all up

haileyschoelkopf reviewed (commented):
Thanks a bunch for creating this and putting the work into it!

Overall, I think having this will be pretty nice for people.

One high-level thing is: we may want to consider the best user experience for this--

1. having people need to opt-in to this, at least at first ( I see that it is opt-in right now): I'm pretty worried about silent errors where someone is editing then rerunning their task but nothing changes. Alternately, having a good fingerprint for the cache that accounts for relevant evaluator or CLI args and the task config or file not having changed might be sufficient to turn this on by default.

2. having a CLI arg or script to reset or delete the cache could be nice. I'm thinking `--cache_outputs` and `--cache_requests` could distinguish our two different caching setups?

inf3rnus commented:
Absolutely! Appreciate you giving the PR a review 🙏

RE:

1. Yup, totally agree! Anyone modifying tasks using this needs to know that they'll need to run the code with the rewrite cache flags. Definitely think the signature idea is a great idea, and also agree that would be necessary to make caching the default behavior. Users will almost certainly run into problems if it's not automagically handled under the hood.

2. Do we want a separate CLI script from the current one, or should the change be added to the existing CLI script?

Digging into the suggestions rn!

inf3rnus commented:
Okay, I've updated everything that was requested, updated the docs as well to include all this info.

`--cache_requests` can now be used via the CLI or via the simple_evaluate func, and accepts the values, `true`, `refresh`, and `delete`, all of which map to what each should do respectively.

For clearing the cache, I've made sure the function that does that does so with some reasonable level of safety in the event the user has set LM_HARNESS_CACHE_PATH to something other than the default, by making sure the file ends with `.pickle`

In addition to this, it has an optional key param for future pickles so we can delineate between them.

Pls lmk if anything else should be changed, I'm not a huge fan of the inline conditions here, but I also don't hate them:

```python
    results = evaluator.simple_evaluate(
        model=args.model,
        model_args=args.model_args,
        tasks=task_names,
        num_fewshot=args.num_fewshot,
        batch_size=args.batch_size,
        max_batch_size=args.max_batch_size,
        device=args.device,
        use_cache=args.use_cache,
        #
        cache_requests=(
            True
            if args.cache_requests == "true" or args.cache_requests == "refresh"
            else False
        ),
        rewrite_requests_cache=True if args.cache_requests == "refresh" else False,
        delete_requests_cache=True if args.cache_requests == "delete" else False,
        #
        limit=args.limit,
        decontamination_ngrams_path=args.decontamination_ngrams_path,
        check_integrity=args.check_integrity,
        write_out=args.write_out,
        log_samples=args.log_samples,
        gen_kwargs=args.gen_kwargs,
        task_manager=task_manager,
        predict_only=args.predict_only,
    )
```

haileyschoelkopf commented:
Hey @inf3rnus , thanks a bunch for this! Sorry for the slow turnaround on my end--been trying to make the ACL deadline with a paper. I will review this end-of-tomorrow at latest!

inf3rnus commented:
Np np, no rush, ready to make the changes whenever you're ready!

haileyschoelkopf commented on .gitignore:
```diff
@@ -16,3 +16,6 @@ temp
 profile_default/
 ipython_config.py
+
+Old/lm_eval
```
remove `Old/lm_eval` from gitignore here?

inf3rnus replied:
Fixed

haileyschoelkopf commented on lm_eval/__main__.py:
```diff
@@ -93,6 +93,13 @@ def parse_eval_args() -> argparse.Namespace:
         metavar="DIR",
         help="A path to a sqlite db file for caching model responses. `None` if not caching.",
     )
+    parser.add_argument(
```
I like the UX here!

inf3rnus replied:
thank you!

haileyschoelkopf commented on lm_eval/evaluator.py:
```diff
@@ -50,7 +58,7 @@ def simple_evaluate(
         Name of model or LM object, see lm_eval.models.get_model
-    :param model_args: Optional[str]
+    :param model_args: Optional[str, dict]
         String arguments for each model class, see LM.create_from_arg_string.
```
```suggestion
        String or dict arguments for each model class, see LM.create_from_arg_string and LM.create_from_arg_object.
```

inf3rnus replied:
Updated

haileyschoelkopf commented on lm_eval/api/task.py:
```diff
@@ -348,20 +356,71 @@ def doc_to_text(self, doc):
+        # used with caching
+        og_limit = limit
+
+        cache_key = f"requests-{task_name}"
```
as mentioned in the other comment--perhaps we should base64-encode this or something, so that the odds of file collisions / deleting undesired files is minimized.

inf3rnus replied:
I've added this in the recent commit, lmk if it looks good to you! Can see how it potentially wasn't safe enough

haileyschoelkopf commented on lm_eval/caching/cache.py:
```diff
@@ -0,0 +1,46 @@
+
+
+# NOTE the "key" param is to allow for flexibility
+def delete_cache(key: str = ""):
```
I wonder whether `delete_cache` might be better to 
1. only delete the requests for the tasks you are running with
2. need to take a key, and as key / filename use the base64 encoding or something of the taskname? would want to minimize any risk of delete_cache() deleting user files, although I think this is already pretty good with the default location being caching/.cache

inf3rnus replied:
I'd be glad to add this in a subsequent batch of work, but I think the current speed up is already a big improvement. In the meantime, users can just manually delete the files they want to refresh in their code editor

#2 is now taken care of!

haileyschoelkopf commented on lm_eval/evaluator.py:
```diff
@@ -109,14 +127,26 @@ def simple_evaluate(
-            },
-        )
+
+        elif type(model_args) == dict:
```
```suggestion
        elif isinstance(model_args, dict):
```

haileyschoelkopf commented on lm_eval/evaluator.py:
```diff
@@ -550,7 +595,10 @@ def evaluate(
                                 ) / (total_size + current_size)
                                 # $$s_z^2 = \frac{(n-1) s_x^2 + (m-1) s_y^2}{n+m-1} + \frac{nm(\bar x - \bar y)^2}{(n+m)(n+m-1)}.$$
-                                if var_score == "N/A" or results[group][stderr] == "N/A":
+                                if (
```
will have to merge into main, as the variance/ stderr calc above has changed (this shouldn't affect your PR though, just merge conflicts with these formatting changes)

inf3rnus replied:
Roger roger, lmk if anything needs changing on my end

haileyschoelkopf commented on test_script.py:
```diff
@@ -0,0 +1,80 @@
+
+# Used to specify alternate cache path, useful if run in a docker container
+# NOTE raw datasets will break if you try to transfer the cache from your host to an image
+LM_HARNESSS_CACHE_PATH = os.getenv("LM_HARNESSS_CACHE_PATH")
```
```suggestion
LM_HARNESS_CACHE_PATH = os.getenv("LM_HARNESS_CACHE_PATH")
```

haileyschoelkopf commented on test_script.py:
```diff
@@ -0,0 +1,80 @@
+    return eval_data
+
+
+if __name__ == "__main__":
```
Thanks for adding this test script! Would you be willing to add a variant of this to the `tests/` dir for use in our pytest unit tests?

Ideally we'd also have a unit test for `delete_cache()`, `save_to_cache()`, and `load_from_cache()` too if possible, plus potentially a unit test for `Task.build_all_requests()`.

inf3rnus replied:
Absolutely! I'll got ahead and add it, and then build a unit test for the caching functionality.

Not entirely sure how I'd write one for Task.build_all_requests(), what do you have in mind for it?

inf3rnus replied:
Alright, it's been moved over into its own script.

I believe I didn't make any changes to Task.build_all_requests(), please correct me if I'm wrong, but may be interested in adding such tests if we end up adding caching to task building as well/might do it for fun later anyway.

The unit tests are now added, and I've confirmed all tests pass on my machine

haileyschoelkopf reviewed (changes requested):
Thank you very much @inf3rnus , this PR mostly looks good to me--just left some comments + suggestions regarding deletion behavior and possibly adding testing!

inf3rnus commented:
Okie dokie, I think we're nearing the finish line. I believe I've addressed all the feedback concerns.

The important thing is tests are passing and the potential collision problem with file deletion is resolved.

Please lmk if anything else needs changing.

Many thanks,
Aaron

haileyschoelkopf commented on lm_eval/caching/cache.py:
```diff
@@ -0,0 +1,61 @@
+PATH = OVERRIDE_PATH if OVERRIDE_PATH else f"{MODULE_DIR}/.cache"
+
+# this is to effectively eliminate the possibility of a cache collision
+random.seed("EleutherAI")
```
ah, actually--we don't want to be modifying the random seed for `random` here, as we need to respect #1412 .

inf3rnus replied:
I just went ahead and removed the rand() seed, it was overkill anyway.

LMK if this is ready for a merge! Got all the tests and lint checker to pass on github :)

haileyschoelkopf replied:
Sorry for letting this sit, really appreciate the contribution! Merging now!

haileyschoelkopf reviewed (commented):
Fixed merge conflicts, I think! left comment on one thing I realized re: not wanting lm_eval to have side effects on random seeds.

haileyschoelkopf commented on lm_eval/evaluator.py:
```diff
@@ -19,17 +22,26 @@
 def simple_evaluate(
     model,
-    model_args: Optional[str] = None,
+    model_args: Optional[str, dict, None] = None,
```
```suggestion
    model_args: Optional[Union[str, dict, None]] = None,
```

Wongboo commented:
using `--cache_requests true` results 
```
 ValueError: not enough values to unpack (expected 2, got 0)
```
for multiple datasets.

WenjinW commented:
I encountered the same problem too. @inf3rnus

inf3rnus commented:
I'll take a look at this tomorrow 🫡, not sure exactly what's going on, but @WenjinW if you have a list of datasets for where this was a problem, that'd be helpful for my debugging

inf3rnus commented:
@Wongboo If you could also please provide me with a datasets where this was a problem I'd be forever grateful 🙏

haileyschoelkopf commented:
Also was experiencing this, seemingly on most things I tried, but hadn't tracked down a cause yet--definitely occurred on `mmlu` and `gsm8k` iirc.

WenjinW commented:
> I'll take a look at this tomorrow 🫡, not sure exactly what's going on, but @WenjinW if you have a list of datasets for where this was a problem, that'd be helpful for my debugging

@inf3rnus Thank you for your prompt reply. TruthfulQA and ARC-C.

Wongboo commented:
I encounter this issue for mmlu shot=5, arc_challenge shot=25, hellaswag shot=10

Wongboo commented:
> using `--cache_requests true` results
> 
> ```
>  ValueError: not enough values to unpack (expected 2, got 0)
> ```
> 
> for multiple datasets after running loglikelihood requests. Full traceback
> 
> ```
> Traceback (most recent call last):                                                                                                                                                                                                                                                                                      
>   File "/home/wjb/micromamba/envs/new_model/bin/lm_eval", line 8, in <module>                                                                                                                                                                                                                                           
>     sys.exit(cli_evaluate())                                                                                                                                                                                                                                                                                            
>              ^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                                             
>   File "/home/wjb/micromamba/envs/new_model/lib/python3.12/site-packages/lm_eval/__main__.py", line 318, in cli_evaluate                                                                                                                                                                                                
>     results = evaluator.simple_evaluate(                                                                                                                                                                                                                                                                                
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                                
>   File "/home/wjb/micromamba/envs/new_model/lib/python3.12/site-packages/lm_eval/utils.py", line 288, in _wrapper                                                                                                                                                                                                       
>     return fn(*args, **kwargs)                                                                                                                                                                                                                                                                                          
>            ^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                                          
>   File "/home/wjb/micromamba/envs/new_model/lib/python3.12/site-packages/lm_eval/evaluator.py", line 230, in simple_evaluate                                                                                                                                                                                            
>     results = evaluate(                                                                                                                                                                                                                                                                                                 
>               ^^^^^^^^^                                                                                                                                                                                                                                                                                                 
>   File "/home/wjb/micromamba/envs/new_model/lib/python3.12/site-packages/lm_eval/utils.py", line 288, in _wrapper                                                                                                                                                                                                       
>     return fn(*args, **kwargs)                                                                                                                                                                                                                                                                                          
>            ^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                                          
>   File "/home/wjb/micromamba/envs/new_model/lib/python3.12/site-packages/lm_eval/evaluator.py", line 402, in evaluate                                                                                                                                                                                                   
>     metrics = task.process_results(                                                                                                                                                                                                                                                                                     
>               ^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                                     
>   File "/home/wjb/micromamba/envs/new_model/lib/python3.12/site-packages/lm_eval/api/task.py", line 1170, in process_results                                                                                                                                                                                            
>     return self.config.process_results(doc, results)                                                                                                                                                                                                                                                                    
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                    
>   File "/home/wjb/micromamba/envs/new_model/lib/python3.12/site-packages/lm_eval/tasks/truthfulqa/utils.py", line 8, in process_results_mc2                                                                                                                                                                             
>     lls, is_greedy = zip(*results)                                                                                                                                                                                                                                                                                      
>     ^^^^^^^^^^^^^^                                                                                                                                                                                                                                                                                                      
> ValueError: not enough values to unpack (expected 2, got 0)     
> ```

I repeat the experiment, notice that this error seems to appear at the last of log likelihood  requests, might be a error from border process.

inf3rnus commented:
@Wongboo 'preciate the info! Could you please tell me how many items in each dataset are you running?

inf3rnus commented:
O yeah, one other thing that would be useful is if you guys could give me the commands you're running, currently cannot get mmlu to produce the bug with a dataset limit of 9 and fewshot set to 10. Also could not replicate with running the full dataset... Currently seeing if the other datasets ya'll mentioned will get this elusive problem to present itself

inf3rnus commented:
Alright, so I reinstalled the package in a brand new conda environment and I'm unable to replicate the problem for any of the datasets listed...

I tried each of these cli invocations to no avail in reproducing the issue.

```
lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks arc_challenge \
    --device cuda:0 \
    --cache_requests true \
    --num_fewshot 25
```

```
lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks gsm8k \
    --device cuda:0 \
    --cache_requests true \
    --num_fewshot 10
```

```
lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks hellaswag \
    --device cuda:0 \
    --cache_requests true \
    --num_fewshot 10
```

```
lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks truthfulqa \
    --device cuda:0 \
    --cache_requests true \
    --num_fewshot 

```

```
lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks mmlu \
    --device cuda:0 \
    --cache_requests true \
    --num_fewshot 5
```

Datasets tested so far:

```
mmlu
gsm8k
arc_challenge
hellaswag
truthfulqa
```
I cannot get it to replicate. Ran the full datasets on all of them, also tested it with the --limit arg in the mml command above with values 100, 10, 9, 5, and 1. (Thought was maybe there's an indexing issue with the grouping of instances potentially, odd/even divisors, maybe a different base in counting might sus out the issue. Although looking at the code it shouldn't influence that.)

Also ran it through the debugger with mmlu and a few different values for the limit arg. And also did the same with eleuther's hf pythia 170m model. Still no dice 🤔

Also just sanity checking, is everyone on the latest commit and has deleted the cache and tried it again? (If any of the interfaces specific to tasks have changed and you ran the cache on an older commit and are now running it on a newer commit, it will def throw. We might want that auto-purging behavior when tasks change, if the heavens can bless me with time!.)

So yeah thought is that if everyone can share the command they're running this will hasten the debugging. Any other spare info likes OS would be nice too. Unlikely the cause, but could be a problem.

Wongboo commented:
Interestingly, I rerun experiment
I delete the cache and rerun the first command you provide twice, issue disappear!
I delete the cache and rerun the first command you provide twice with `acclerate launch`, issue appear!
I delete the cache and rerun the first command you provide twice with `acclerate launch` in the second run, issue disappear!
I delete the cache and rerun the first command you provide twice with `acclerate launch` in the first run, issue disappear!
I suggest you rerun experiment with `acclerate launch`

Wongboo commented:
Another thing is to notice is even adding`--cache_requests true`. program may hang on 
 https://github.com/EleutherAI/lm-evaluation-harness/blob/3bdf25ec00ba3be6fbce88361080cb31aff069f4/lm_eval/evaluator.py#L194 a very long time by triggering `load_dataset`. 
It seems that when using `--cache_requests true`, we only need to know `task._config.task` for `cache_key` and its hierachy, no need to load the dataset.

inf3rnus commented:
> Interestingly, I rerun experiment I delete the cache and rerun the first command you provide twice, issue disappear! I delete the cache and rerun the first command you provide twice with `acclerate launch`, issue appear! I delete the cache and rerun the first command you provide twice with `acclerate launch` in the second run, issue disappear! I delete the cache and rerun the first command you provide twice with `acclerate launch` in the first run, issue disappear! I suggest you rerun experiment with `acclerate launch`

Apperciate the info @Wongboo 🙏

By any chance could you also provide the exact command? I just want to make sure we leave nothing up to doubt.

I'm running this, and I can't get it to reproduce the problem :/

```
accelerate launch -m lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks mmlu \
    --device cuda:0 \
    --cache_requests true \
    --num_fewshot 5 \
    --limit 5
```

Wongboo commented:
```
accelerate launch -m lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks arc_challenge \
    --cache_requests true \
    --num_fewshot 25
```
delete the cache and running twice, problem appears, you should not specify `--device cuda:0`?
I'm still wondering why we have to load the dataset even if cache exists and `--cache_requests true`.

haileyschoelkopf commented:
I've found a fix for this for now, in #1573 : having each rank save a cache dependent on rank and worldsize fixes the issues I was experiencing.  

```
accelerate launch -m lm_eval --model hf \
    --model_args pretrained=EleutherAI/pythia-70m,dtype="float" \
    --tasks arc_challenge \
    --cache_requests true \
    --num_fewshot 25
delete the cache and running twice, problem appears
```
to replicate the problem/fix.

Wongboo commented:
But still, I'm wondering why we have to load the dataset even if cache exists and `--cache_requests true`.

haileyschoelkopf commented:
Plan to address this via #1558 if possible.

sigridjineth commented:
@haileyschoelkopf I am having the same issue when running like this.

```
(.venv) sionic@iZmj7ir0ircgij46j89st9Z:~/sigrid/ko-lm-evaluation-harness$ lm_eval --model hf     --model_args OrionStarAI/Orion-14B-Base     --tasks haerae,kmmlu     --batch_size 8 --cache_requests true
2024-03-20:12:45:39,193 INFO     [__main__.py:251] Verbosity set to INFO
2024-03-20:12:45:43,016 INFO     [__main__.py:335] Selected Tasks: ['haerae', 'kmmlu']
2024-03-20:12:45:43,016 INFO     [__main__.py:336] Loading selected tasks...
2024-03-20:12:45:43,018 INFO     [evaluator.py:131] Setting random seed to 0 | Setting numpy seed to 1234 | Setting torch manual seed to 1234
Traceback (most recent call last):
  File "/home/sionic/.venv/bin/lm_eval", line 8, in <module>
    sys.exit(cli_evaluate())
  File "/home/sionic/sigrid/lm-evaluation-harness/lm_eval/__main__.py", line 342, in cli_evaluate
    results = evaluator.simple_evaluate(
  File "/home/sionic/sigrid/lm-evaluation-harness/lm_eval/utils.py", line 288, in _wrapper
    return fn(*args, **kwargs)
  File "/home/sionic/sigrid/lm-evaluation-harness/lm_eval/evaluator.py", line 164, in simple_evaluate
    lm = lm_eval.api.registry.get_model(model).create_from_arg_string(
  File "/home/sionic/sigrid/lm-evaluation-harness/lm_eval/api/model.py", line 132, in create_from_arg_string
    args = utils.simple_parse_args_string(arg_string)
  File "/home/sionic/sigrid/lm-evaluation-harness/lm_eval/utils.py", line 73, in simple_parse_args_string
    args_dict = {
  File "/home/sionic/sigrid/lm-evaluation-harness/lm_eval/utils.py", line 74, in <dictcomp>
    k: handle_arg_string(v) for k, v in [arg.split("=") for arg in arg_list]
ValueError: not enough values to unpack (expected 2, got 1)
(.venv) sionic@iZmj7ir0ircgij46j89st9Z:~/sigrid/ko-lm-evaluation-harness$ lm_eval --model hf     --model_args OrionStarAI/Orion-14B-Base     --tasks haerae,kmmlu     --batch_size 8 --cache_requests true
```

haileyschoelkopf commented:
Hi @sigridjineth , I believe you need to pass ` --model_args pretrained=OrionStarAI/Orion-14B-Base` rather than `--model_args OrionStarAI` !
